server/src/services/paygov.es6

```diff
@@ -23,125 +24,102 @@ paygov.createToken = (permitId) => {
+  }); 
 };
 
 /**
- * @function createSuccessUrl - create success url for paygov request
+ * @function returnUrl - create return url for paygov request
  * @param {string} forestAbbr - forest abbreviation
  * @param {string} permitId - permit id
+ * @param {Boolean} isCancelUrl - whether to include the cancel query
  * @return {string} - success URL for payGov
  */
-paygov.createSuccessUrl = (forestAbbr, permitId) => {
-  const token = paygov.createToken(permitId);
-  return `${
-    vcapConstants.INTAKE_CLIENT_BASE_URL
-  }/christmas-trees/forests/${forestAbbr}/applications/permits/${permitId}?t=${token}`;
-};
+paygov.returnUrl = (forestAbbr, permitId, isCancelUrl) => {
+  let cancelQuery = '';
+  let completeRoute = '/permits';
+  if (isCancelUrl) {
+    cancelQuery = 'cancel=true&';
+    completeRoute = '';
+  }
+  return new Promise((resolve, reject) =>{
+    paygov.createToken(permitId)
+      .then(token => {
+        resolve(`${
+          vcapConstants.INTAKE_CLIENT_BASE_URL
+        }/christmas-trees/forests/${forestAbbr}/applications${completeRoute}/${permitId}?${cancelQuery}t=${token}`
+        );
+      })
+      .catch(error => reject(error));
+  });
 
-/**
- * @function createCancelUrl - create cancel url for paygov request
- * @param {string} forestAbbr - forest abbreviation
- * @param {string} permitId - permit id
- * @return {string} - cancel URL for payGov
- */
-paygov.createCancelUrl = (forestAbbr, permitId) => {
-  const token = paygov.createToken(permitId);
-  return `${
-    vcapConstants.INTAKE_CLIENT_BASE_URL
-  }/christmas-trees/forests/${forestAbbr}/applications/${permitId}?cancel=true&t=${token}`;
 };
 
 /**
- * @function getXmlForToken - Generate XML from the template to use for getting pay.gov transaction token.
+ * @function getXmlStartCollection - Generate XML from the template to use for getting pay.gov transaction token.
  * @param {string} forestAbbr - forest abbreviation
  * @param {string} possFinancialId - forest's financial id
  * @param {Object} permit - permit object from database
  * @return {string} - XML for payGov startOnlineCollection request
  */
-paygov.getXmlForToken = (forestAbbr, possFinancialId, permit) => {
+paygov.getXmlStartCollection = (forestAbbr, possFinancialId, permit) => {
   const tcsAppID = vcapConstants.PAY_GOV_APP_ID;
-  let url_success = '';
-  try {
-    url_success = paygov.createSuccessUrl(forestAbbr, permit.permitId);
-  } catch (e) {
-    logger.error('problem creating success url for permit ' + permit.id, e);
-  }
-
-  let url_cancel;
-  try {
-    url_cancel = paygov.createCancelUrl(forestAbbr, permit.permitId);
-  } catch (e) {
-    logger.error('problem creating success url for permit ' + permit.id, e);
-  }
-
-  const xmlTemplate = [
-    {
-      'soap:Envelope': [
-        {
-          _attr: {
-            'xmlns:soap': 'http://schemas.xmlsoap.org/soap/envelope/'
-          }
-        },
-        {
-          'soap:Body': [
+  return new Promise((resolve, reject) => {
```

Comment:
Same here, i don't think you need to create another promise, just return what you need to in the `then` and don't catch the error.